### PR TITLE
feat(*): upgrade Frequency Mainnet to 6-Second Block Time

### PR DESCRIPTION
**Pull request by enddynayn**
This update reduces block time to 6 seconds. As a result, the Capacity Epoch length has been adjusted to account for the reduced latency. Additionally, the default interval sealing time has been changed to 6 seconds.

A migration has also been added to update the StorageVersion of the CollatorSelection pallet.

Commands to test the runtime migration:

Paseo
```
make try-runtime-create-snapshot-paseo-testnet
make try-runtime-use-snapshot-paseo-testnet
```

Mainnet
```
cargo build --features frequency,try-runtime&& \
try-runtime --runtime ./target/release/wbuild/frequency-runtime/frequency_runtime.wasm on-runtime-upgrade live --uri wss://1.rpc.frequency.xyz:443 --pallet Capacity --pallet System
```

Or

Paseo
```
cargo build --features frequency,try-runtime && \
try-runtime --runtime ./target/release/wbuild/frequency-runtime/frequency_runtime.wasm on-runtime-upgrade live --uri wss://1.rpc.frequency.xyz:443 --pallet Capacity --pallet System
```

```
cargo build --features frequency-testnet,try-runtime && \
try-runtime --runtime ./target/debug/wbuild/frequency-runtime/frequency_runtime.wasm on-runtime-upgrade live --uri wss://rpc.testnet.amplica.io:443 -pallet  Capacity --pallet
```

Screenshot 2024-10-07 at 3 13 25 PM

#1920 
#2141

### CODE REVIEW

**Review by aramikm — APPROVED** [review]
I've only read the code but didn't execute the try runtimes.
It looked good to me!

File: runtime/common/src/constants.rs [hunk 1]
```diff
@@ -350,7 +338,7 @@ pub type MaxPaginatedPageId = ConstU16<32>;
 /// The maximum number of actions in itemized actions
 pub type MaxItemizedActionsCount = ConstU32<5>;
 /// The number of blocks for Stateful mortality is 24 hours
```
Comment (mattheworris) [inline runtime/common/src/constants.rs:340]: nit: This comment should be 48 hours to match the value below.

**Review by shannonwells — COMMENTED** [review]
The CapacityEpoch should still be one day?

File: runtime/common/src/constants.rs [hunk 2]
```diff
@@ -397,7 +385,7 @@ impl sp_std::fmt::Debug for MaxItemizedBlobSizeBytes {
 pub type CapacityMinimumStakingAmount = ConstU128<{ currency::EXISTENTIAL_DEPOSIT }>;
 pub type CapacityMinimumTokenBalance = ConstU128<{ currency::DOLLARS }>;
 pub type CapacityMaxUnlockingChunks = ConstU32<4>;
-pub type CapacityMaxEpochLength = ConstU32<7_200>; // one day, assuming 12 second blocks.
+pub type CapacityMaxEpochLength = ConstU32<{ 2 * DAYS }>; // Two days, assuming 6 second blocks.
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply runtime/common/src/constants.rs:388]: Great question! The move from a 1-day to a 2-day epoch is mainly for future-proofing. Just to clarify, this 2-day setting represents the maximum, not the actual epoch length. The actual epoch is still set to 1 day, as shown in the migration. By setting the max to 2 days, we allow for future increases via governance without needing code changes.

File: runtime/common/src/fee.rs [hunk 3]
```diff
@@ -59,7 +59,7 @@ mod test {
 		let full_block = WeightToFee::weight_to_fee(&MAXIMUM_BLOCK_WEIGHT);
 		// A bounded assertion to consider changes in generated extrinsic base weight.
 		assert!(full_block >= 2 * 150 * CENTS);
-		assert!(full_block <= 10 * DOLLARS);
+		assert!(full_block <= 80 * DOLLARS);
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply runtime/common/src/fee.rs:62]: The throughput has been increased by 8x, which means the cost to purchase a whole block has also gone up accordingly. As a result, the test failed correctly, so I’ve updated it to reflect the new approximate cost for purchasing a block.

File: runtime/common/src/constants.rs [hunk 4]
```diff
@@ -92,7 +85,7 @@ pub type MsaMaxPublicKeysPerMsa = ConstU8<25>;
 /// The maximum size of the provider name (in bytes)
 pub type MsaMaxProviderNameSize = ConstU32<16>;
 /// The number of blocks per virtual bucket
-pub type MSAMortalityWindowSize = ConstU32<100>;
+pub type MSAMortalityWindowSize = ConstU32<{ 20 * MINUTES }>;
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply runtime/common/src/constants.rs:88]: They should be equivalent, but measured in terms of blocks over time. For example:
```
100 * 12 sec/ 60
```
This calculates to 20 minutes, with the measurement based on the number of blocks.

File: scripts/init.sh [hunk 5]
```diff
@@ -136,7 +136,7 @@ start-frequency-instant)
   ;;
 
 start-frequency-interval)
-  defaultInterval=12
+  defaultInterval=6
```
Comment (wilwade) [inline scripts/init.sh:139]: Do we want to also change the default for the `sealing_interval` parameter? (doesn't have to be this PR)

File: pallets/capacity/src/lib.rs [hunk 6]
```diff
@@ -88,7 +88,7 @@ pub mod pallet {
 	}
 
 	/// the storage version for this pallet
-	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(3);
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(4);
```
Comment (wilwade) [inline pallets/capacity/src/lib.rs:91]: ❤️

**Review by wilwade — APPROVED** [review]
One optional change, but otherwise looks great!

- Reviewed code
- Pulled and searched for any remaining usages of `12` or `7200`

File: runtime/common/src/fee.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (aramikm) [reply runtime/common/src/fee.rs:62]: 8x does not make sense. It should be 4x. The reason is that we increase the block processing capacity from 0.5 seconds to 2 seconds. This means each block can process 4 times more. So this should really be 4 times. The throughput is 8x becasue we generate 2 blocks in the same timestamp but here we are testing just 1 block as far as I can tell.

File: scripts/init.sh (hunk 5 above)
Comment (enddynayn) [reply scripts/init.sh:139]: Good idea! I update the default to 6 seconds.
```
#[clap(long, help = "The interval in seconds", default_value = "6", value_name = "SECONDS")]
```

File: runtime/common/src/constants.rs (hunk 1 above)
Comment (enddynayn) [reply runtime/common/src/constants.rs]: Good point!

**Review by mattheworris — APPROVED** [review]
- Read through the changes
- Non-blocking comments
🚢 it!

**Review by saraswatpuneet — APPROVED** [review]
Read through the changes and comments
Looks all good. Would try this new image soon 🚢

**Pull request merged**